cli/types/index.d.ts

```diff
@@ -174,757 +182,739 @@ declare namespace Cypress {
+    screenshot(fileName: string, options?: Partial<Loggable & Timeoutable>): Chainable<null>
+
+    /**
+     * Scroll an element into view.
+     *
+     * @see https://on.cypress.io/scrollintoview
+     */
+    scrollIntoView(options?: Partial<ScrollIntoViewOptions>): Chainable<Subject>
 
     /**
+     * Scroll to a specific position.
+     *
+     * @see https://on.cypress.io/scrollto
+     */
+    scrollTo(position: PositionType, options?: Partial<ScrollToOptions>): Chainable<Subject>
+    scrollTo(x: number | string, y: number | string, options?: Partial<ScrollToOptions>): Chainable<Subject>
+
+    /**
+     * Select an `<option>` within a `<select>`.
+     *
      * @see https://on.cypress.io/select
      */
-    select(text: string | string[], options?: SelectOptions): Chainable
-    select(value: string | string[], options?: SelectOptions): Chainable
+    select(text: string | string[], options?: Partial<SelectOptions>): Chainable<Subject>
+    select(value: string | string[], options?: Partial<SelectOptions>): Chainable<Subject>
 
     /**
+     * Start a server to begin routing responses to `cy.route()` and `cy.request()`.
+     *
      * @see https://on.cypress.io/server
      */
-    server(options?: ServerOptions): Chainable
+    server(options?: Partial<ServerOptions>): Chainable<null>
 
     /**
+     * Set a browser cookie.
+     *
      * @see https://on.cypress.io/setcookie
      */
-    setCookie(name: string, value: string, options?: SetCookieOptions): Chainable
+    setCookie(name: string, value: string, options?: Partial<SetCookieOptions>): Chainable<Cookie>
 
     /**
+     * Create an assertion. Assertions are automatically retried until they pass or time out.
+     *
      * @see https://on.cypress.io/should
      */
-    should(chainers: string, value?: any): Chainable
-    should(chainers: string, method: string, value: any): Chainable
-    should(fn: (currentSubject?: any) => void): Chainable
+    should: Chainer<Subject>
 
     /**
+     * Get sibling DOM elements.
+     *
      * @see https://on.cypress.io/siblings
      */
-    siblings(options?: LoggableTimeoutable): Chainable
-    siblings(selector: string, options?: LoggableTimeoutable): Chainable
+    siblings<K extends keyof HTMLElementTagNameMap>(selector: K, options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<HTMLElementTagNameMap[K]>>
+    siblings<E extends Node = HTMLElement>(options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<E>>
+    siblings<E extends Node = HTMLElement>(selector: string, options?: Partial<Loggable & Timeoutable>): Chainable<JQuery<E>>
 
     /**
+     * Expand an array into multiple arguments.
+     * Identical to `.then()`, but always expects an array-like structure as its subject.
+     *
      * @see https://on.cypress.io/spread
      */
-    spread(fn: (...args: any[]) => any): Chainable
+    spread(fn: (...args: any[]) => void): Chainable<Subject> // Can't type this any better without breaking up Chainable

```

Comment:
These types aren't quite right. I had them working with commands being broken down. Without that, I don't have a way of typing a `Subject` that is an array as just a Subject again.

I may have to either add a `ChainableArray` interface or leave anything hitting `.spread` as `any`